datasafe-rest/src/main/java/de/adorsys/datasafe/rest/controller/DocumentController.java

```diff
@@ -0,0 +1,86 @@
+package de.adorsys.datasafe.rest.controller;
+
+import de.adorsys.datasafe.business.api.types.UserID;
+import de.adorsys.datasafe.business.api.types.UserIDAuth;
+import de.adorsys.datasafe.business.api.types.action.ListRequest;
+import de.adorsys.datasafe.business.api.types.action.ReadRequest;
+import de.adorsys.datasafe.business.api.types.action.RemoveRequest;
+import de.adorsys.datasafe.business.api.types.action.WriteRequest;
+import de.adorsys.datasafe.business.api.types.keystore.ReadKeyPassword;
+import de.adorsys.datasafe.business.api.types.resource.DefaultPrivateResource;
+import de.adorsys.datasafe.business.api.types.resource.PrivateResource;
+import de.adorsys.datasafe.business.impl.service.DefaultDatasafeServices;
+import lombok.SneakyThrows;
+import org.springframework.util.StreamUtils;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.springframework.http.MediaType.APPLICATION_OCTET_STREAM_VALUE;
+
+@RestController
+@RequestMapping("/document")
+public class DocumentController {
+
+    private final DefaultDatasafeServices dataSafeService;
+
+    public DocumentController(DefaultDatasafeServices dataSafeService) {
+        this.dataSafeService = dataSafeService;
+    }
+
+    @SneakyThrows
+    @GetMapping(produces = APPLICATION_OCTET_STREAM_VALUE)
+    public void readDocument(@RequestParam String user,
```

Comment:
Albeit it is just for testing, still it is better to follow best practices since change is trivial
Please send user and password as headers

https://www.owasp.org/index.php/Information_exposure_through_query_strings_in_url
https://security.stackexchange.com/questions/29598/should-sensitive-data-ever-be-passed-in-the-query-string

Reply:
ok